### ISSUE TITLE
`let` in `$lookup` aggregation

### ISSUE THREAD

**Issue by guludo** [mongomock/mongomock#536]
Hello everyone!

Currently MongoMock does not support using `let` in `$lookup`, and it gracefully lets the user know with the following message:

```
NotImplementedError: Although 'let' is a valid lookup operator for the aggregation pipeline, it is currently not implemented in Mongomock.
```

I'm creating this issue so that people (like myself) interested in this feature might get notifications regarding this feature, in case it is added in the future. :-)

Regards!

**Comment by pcorpet** [comment]
It probably requires #535 first.

**Comment by MoustafaWehbe** [comment]
I am getting the issue any solution ??? @pcorpet

**Comment by mchfrnc** [comment]
Hello @pcorpet, is there any chance for this to be implemented soon?
Thank you!

**Comment by pcorpet** [comment]
Thanks for pushing, I'm almost there for #535 which will is a big step for implementing `let`

**Comment by lmoretto** [comment]
Hello @pcorpet 
I have the same issue. I just tried the latest develop version, which includes https://github.com/mongomock/mongomock/pull/698, but I still get an error when using `$let` in a `$lookup` stage (https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/#join-conditions-and-uncorrelated-sub-queries):
```
NotImplementedError: Although 'let' is a valid lookup operator for the aggregation pipeline, it is currently not implemented in Mongomock.
```

**Comment by pcorpet** [comment]
Yes, the `let` + `pipeline` in `$lookup` is not implemented yet. What was added is the `$let` operator (https://docs.mongodb.com/manual/reference/operator/aggregation/let/).

There is still some more work to make the pipeline step starts another pipeline although it should be doable now.

**Comment by jjaviergalvez** [comment]
Hi guys. Thanks for this amazing library. Any update on this?

**Comment by cvmartin** [comment]
I can confirm that the branch `lookup-let` fixes the issue, at least for me. Looking forward to see it released.
(edited)

**Comment by abhikhedekar4241** [comment]
Wouldn't merging `lookup-let` fix this?

**Comment by tarekyz** [comment]
Hey @pcorpet, the branch `lookup-let` fixes the issue really well. Do you have plans on releasing this?

**Comment by dmxlol** [comment]
@pcorpet hello, do you have any plans to release lookup-let features?
